lucene/core/src/test/org/apache/lucene/search/TestTopDocsCollector.java

```diff
@@ -404,44 +424,288 @@ public void testTotalHits() throws Exception {
 
   public void testGlobalScore() throws Exception {
     Directory dir = newDirectory();
-    RandomIndexWriter writer = new RandomIndexWriter(random(), dir);
-    try (LineFileDocs docs = new LineFileDocs(random())) {
```

Comment:
Hmm, could we also let the `LineFileDocs` based test remain, in addition to the new tests? I find it to be a good way to do random tests on realistic(ish) data sets.

Reply:
I restored it in https://github.com/apache/lucene-solr/pull/904/commits/f55633c8812d6e33ffab309f1f6d8af53e56b774